cfpb/cfgov-refresh
# Improve notifier/validation plugins

Improves the notifier and validation plugins.

## Additions

- Added an insertTarget option to allow the notification to be inserted to an elem 
  other than the one that triggered it (typically a form).
- Added an insertLocation option to allow the notification to be appended rather 
  than prepended to the targeted element

## Removals

- None

## Changes

- Cleaned up all three JS files to reduce complexity, organize functions, and remove almost all the warnings

## Testing

- checkout branch
- use the subscription form at `/blog/`
- use the filter form at `/the-bureau/leadership-calendar/`
- everything should be the same

## Review

- @sebworks 
- @KimberlyMunoz 
- @anselmbradford 

## Preview

![screen shot 2015-05-20 at 3 39 38 pm](https://cloud.githubusercontent.com/assets/1280430/7734781/7f088ce6-ff06-11e4-800e-8f70b877d5fd.png)


[Preview this PR without the whitespace changes](?w=0)

## Notes

- I ignored the characters per line and underscore in the plugin name errors
- I couldn't reduce the complexity of the last three items, but they're better than they started so I'm considering it a win
- Although a screenshot is included, this PR doesn't demo the notifications inserted outside the default, but that can be tested locally if interested
